src/lib/y2storage/encryption_method.rb

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2storage/encryption_processes/luks1"
+require "y2storage/encryption_processes/swap"
+
+module Y2Storage
+  # Class representing each one of the mechanisms YaST can use to encrypt a
+  # block device.
+  #
+  # Not to be confused with {EncryptionType} that represents the encryption
+  # technology used underneath.
+  #
+  # Each encryption method has a name and encapsulates the corresponding
+  # steps and encryption properties (e.g. the type).
+  class EncryptionMethod
+    include Yast::I18n
+    extend Yast::I18n
+
+    # Constructor
+    #
+    # @param id [Symbol]
+    # @param label [String]
+    # @param process_class [#new] class name of the encryption process (e.g.,
+    #   EncryptionProcesses::Luks1)
+    def initialize(id, label, process_class)
+      textdomain "storage"
+
+      @id = id
+      @label = label
+      @process_class = process_class
+    end
+
+    LUKS1 = new(
+      :luks1, N_("Regular LUKS1"), EncryptionProcesses::Luks1
+    )
+    RANDOM_SWAP = new(
+      :random_swap, N_("Random Swap"), EncryptionProcesses::Swap
+    )
+
+    ALL = [LUKS1, RANDOM_SWAP].freeze
+    private_constant :ALL
+
+    # Sorted list of all possible settings
+    #
+    # @return [Array<Y2Storage::EncryptionMethod>]
+    def self.all
+      ALL.dup
+    end
+
+    # Looks for the encryption method used for the given encryption device
+    #
+    # @param encryption [Y2Storage::Encryption]
+    # @return [Y2Storage::EncryptionMethod, nil]
+    def self.for_device(encryption)
+      all.find { |m| m.used_for?(encryption) }
+    end
+
+    # Looks for the encryption method by its symbol representation
+    #
+    # @param value [#to_sym]
+    # @return [Y2Storage::EncryptionMethod, nil] the encryption method found if any; nil otherwise
+    def self.find(value)
+      all.find { |i| i.is?(value) }
+    end
+
+    # @return [Symbol]
```

Comment:
Please, document it. For example: "name to represent the encryption method (e.g., :luks1, :random_swap)"